include/readdy/kernel/singlecpu/model/SCPUNeighborList.h

```diff
@@ -71,15 +70,18 @@ class CellLinkedList {
     CellLinkedList(data_type &data, const readdy::model::Context &context)
             : _data(data), _context(context), _head{}, _list{}, _radius{0} {};
 
-    void setUp(scalar skin, cell_radius_type radius) {
-        if (!_is_set_up || _skin != skin || _radius != radius) {
-
-            _skin = skin;
+    void setUp(scalar interactionDistance, cell_radius_type radius) {
+        if (!_isSetUp || _interactionDistance != interactionDistance || _radius != radius) {
```

Comment:
I tried this now. I stumbled upon the fact that the NL is owned by the StateModel. So far the NL would be created when StateModel was created, but setup was delayed until `StateModel::initializeNeighborList` was called, which lead to the necessity of the `isSetUp` flags internally (because you don't want to update the NL before it was set up). If now the setup shall happen during NL construction I see three options:
- either to not construct the NL in the StateModel ctor and delay construction until `StateModel::initializeNeighborList` was called. This induces the necessity to check for nullity of the NL unique_ptr (e.g. in `StateModel::updateNeighborList`), which is basically the same thing as the `isSetUp` flags but now it appears higher up (namely in the StateModel)
- or construct the StateModel already with the knowledge of the cutoff distance. Then everything is already set up after the StateModel ctor and the `CreateNeighborList` action becomes obsolete.
- decouple NL and StateModel and have them live side by side in the kernel rather than NL owned by StateModel. I currently don't see all the implications this might have, but it seems most flexible.